code/go/0chain.net/smartcontract/dbs/event/snapshot.go

```diff
@@ -0,0 +1,222 @@
+//allocated - clients have locked up storage by purchasing allocations
+//unallocated - this is equal to (staked - allocated) and allows clients to purchase allocations with free space blobbers.
+//used - this is the actual usage or data that is in the server.
+//staked + unstaked = max_capacity
+//allocated + unallocated = staked
+
+// swagger:model Snapshot
+type Snapshot struct {
+	Round int64 `gorm:"primaryKey;autoIncrement:false" json:"round"`
+
+	TotalMint            int64 `json:"total_mint"`
+	TotalChallengePools  int64 `json:"total_challenge_pools"`  //486 AVG show how much we moved to the challenge pool maybe we should subtract the returned to r/w pools
+	ActiveAllocatedDelta int64 `json:"active_allocated_delta"` //496 SUM total amount of new allocation storage in a period (number of allocations active)
+	ZCNSupply            int64 `json:"zcn_supply"`             //488 SUM total ZCN in circulation over a period of time (mints). (Mints - burns) summarized for every round
+	TotalValueLocked     int64 `json:"total_value_locked"`     //487 SUM Total value locked = Total staked ZCN * Price per ZCN (across all pools)
+	ClientLocks          int64 `json:"client_locks"`           //487 SUM How many clients locked in (write/read + challenge)  pools
+
+	// updated from blobber snapshot aggregate table
+	AverageWritePrice    int64 `json:"average_write_price"`   //*494 AVG it's the price from the terms and triggered with their updates //???
+	TotalStaked          int64 `json:"total_staked"`          //*485 SUM All providers all pools
+	SuccessfulChallenges int64 `json:"successful_challenges"` //*493 SUM percentage of challenges failed by a particular blobber
+	TotalChallenges      int64 `json:"total_challenges"`      //*493 SUM percentage of challenges failed by a particular blobber
+	AllocatedStorage     int64 `json:"allocated_storage"`     //*490 SUM clients have locked up storage by purchasing allocations (new + previous + update -sub fin+cancel or reduceed)
+	MaxCapacityStorage   int64 `json:"max_capacity_storage"`  //*491 SUM all storage from blobber settings
+	StakedStorage        int64 `json:"staked_storage"`        //*491 SUM staked capacity by delegates
+	UsedStorage          int64 `json:"used_storage"`          //*491 SUM this is the actual usage or data that is in the server - write markers (triggers challenge pool / the price).(bytes written used capacity)
+}
+
+type FieldType int
+
+const (
+	Allocated = iota
+	MaxCapacity
+	Staked
+	Used
+)
+
+type AllocationValueChanged struct {
+	FieldType    FieldType
+	AllocationId string
+	Delta        int64
+}
+type AllocationBlobberValueChanged struct {
+	FieldType    FieldType
+	AllocationId string
+	BlobberId    string
+	Delta        int64
+}
+
+func (edb *EventDb) GetDifference(start, end int64, roundsPerPoint int64, row, table string) ([]int64, error) {
+	query := fmt.Sprintf(`
+		SELECT %s - LAG(%s,1, CAST(0 AS Bigint)) OVER(ORDER BY round ASC) 
```

Comment:
why do we use window here, it won't work

Reply:
Did you follow up here https://0chain.slack.com/archives/C03S5PW1DG9/p1662891460016749?thread_ts=1662839975.419349&cid=C03S5PW1DG9